sqlglot/optimizer/qualify_tables.py

```diff
@@ -66,19 +65,9 @@ def qualify_tables(
                     if not source.args.get("catalog"):
                         source.set("catalog", exp.to_identifier(catalog))
 
-                # Unnest joins attached in tables by appending them to the closest query
-                for join in source.args.get("joins") or []:
-                    scope.expression.append("joins", join)
-
-                source.set("joins", None)
-                source.set("wrapped", None)
-
                 if not source.alias:
-                    source = source.replace(
-                        alias(
-                            source, name or source.name or next_alias_name(), copy=True, table=True
-                        )
-                    )
+                    # Mutates source by attaching an alias to it
+                    alias(source, name or source.name or next_alias_name(), copy=False, table=True)
```

Comment:
This change was important because we may have a nested join like `(tbl1 JOIN tbl2)`, where `tbl1` is an ancestor of `tbl2` in the AST. Since we want to treat both tables as sources, we will most likely get a situation where `tbl1` is replaced first and so when the loop visits `tbl2` its reference will be dangling and the alias addition will not be reflected in the resulting AST.

For example, this is what would happen due to this `replace` call:

```
SELECT * FROM (t1 CROSS JOIN t2) --> SELECT * FROM (t1 AS t1 CROSS JOIN t2)
```